File: Linked List/M-328. Odd Even Linked List/M-328. Odd Even Linked List.ts
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var oddEvenList = function (head) {
  let oddHead = new ListNode();
  let oddNode = oddHead;
  let evenHead = new ListNode();
  let evenNode = evenHead;
  let i = 1;
  while (head) {
    if (i % 2 === 1) {
      oddNode.next = head;
      oddNode = oddNode.next;
    } else {
      evenNode.next = head;
      evenNode = evenNode.next;
    }
    head = head.next;
    i++;
  }
  evenNode.next = null;
  oddNode.next = evenHead.next;
  return oddHead.next;
};

File: Linked List/M-328. Odd Even Linked List/M-328. Odd Even Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        odd_head, even_head = ListNode(), ListNode()
        odd_tail, even_tail = odd_head, even_head
        node = head
        count = 0

        while node:
            count += 1
            if count % 2:
                odd_tail.next = ListNode(node.val)
                odd_tail = odd_tail.next
            else:
                even_tail.next = ListNode(node.val)
                even_tail = even_tail.next
            node = node.next

        if count == 1:
            return odd_head.next

        odd_tail.next = even_head.next
        return odd_head.next
